PermanentOrg/web-app
# Public Gallery Landing Page (Static)

This PR adds the Public Gallery Landing Page. It can be accessed at `/gallery`. To test this new page, you'll probably want to add some public archives to the "Featured Archives" section of your environment. To do this, create an array of `FeaturedArchive` objects and convert it to a JSON string to add to your `.env` file under the `FEATURED_ARCHIVES` key.

<details>
<summary>A more in-depth description with example data for you to copy!</summary>
I first start off with this featured archive data:

```json
[
  {
    "name": "The Natalie Martin Archive",
    "type": "type.archive.person",
    "description": "Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum",
    "archiveNbr": "0002-0000"
  },
  {
    "name": "The Gallery Test Archive",
    "type": "type.archive.family",
    "description": "Test test test test test test test!!!",
    "archiveNbr": "0005-0000"
  },
  {
    "name": "The Natalie Martin Archive",
    "type": "type.archive.organization",
    "description": "Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum",
    "archiveNbr": "0002-0000"
  },
  {
    "name": "The Natalie Martin Archive",
    "type": "type.archive.person",
    "description": "Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum",
    "archiveNbr": "0002-0000"
  },
  {
    "name": "The Natalie Martin Archive",
    "type": "type.archive.family",
    "description": "Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum",
    "archiveNbr": "0002-0000"
  },
  {
    "name": "The Natalie Martin Archive",
    "type": "type.archive.organization",
    "description": "Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum",
    "archiveNbr": "0002-0000"
  }
]
```

The `archiveNbr` values must correspond to archives on your local environment with public archives!

I converted this to a compressed JSON string by using `JSON.stringify` in my browser dev tools (though you could also probably find a converter online as well!). From there I put it in my `.env` file like this:
```
FEATURED_ARCHIVES="[{\"name\":\"The Natalie Martin Archive\",\"type\":\"type.archive.person\",\"description\":\"Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum\",\"archiveNbr\":\"0002-0000\"},{\"name\":\"The Gallery Test Archive\",\"type\":\"type.archive.family\",\"description\":\"Test test test test test test test!!!\",\"archiveNbr\":\"0005-0000\"},{\"name\":\"The Natalie Martin Archive\",\"type\":\"type.archive.organization\",\"description\":\"Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum\",\"archiveNbr\":\"0002-0000\"},{\"name\":\"The Natalie Martin Archive\",\"type\":\"type.archive.person\",\"description\":\"Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum\",\"archiveNbr\":\"0002-0000\"},{\"name\":\"The Natalie Martin Archive\",\"type\":\"type.archive.family\",\"description\":\"Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum\",\"archiveNbr\":\"0002-0000\"},{\"name\":\"The Natalie Martin Archive\",\"type\":\"type.archive.organization\",\"description\":\"Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum\",\"archiveNbr\":\"0002-0000\"}]"
```

After doing that, you can re-build or re-launch the dev proxy server (if you're running the dev proxy server you can also just run `npm run secrets` to rebuild secrets while the server is running, but this will not work if you're running through Vagrant). You should see the featured archives on the page now!
</details>

**Steps to test this page:**
* Add in a variety of public archives and see if everything appears correctly.
* Test the page in both logged in/logged out states and verify the buttons work.
* Test the page in mobile and desktop view.

This PR also adds a new type of secret to our `.env` files: "Optional Secrets". This allows us to store Featured Archive data in the `.env` file, without having to make *everyone* and CI/deployment processes update their `.env` files for what is a pretty minor thing. Previously, all secrets were required and would throw errors if they were not defined in `.env`, but any "optional secrets" will be assigned a default value if they're not defined.

Resolves PER-8683: Public Gallery Landing Page (Static).
